src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,68 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  setFilter: React.Dispatch<React.SetStateAction<Filter>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  query: string;
+};
+
+export const TodoFilter: React.FC<Props> = ({ setFilter, setQuery, query }) => {
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case 'All':
+        setFilter(Filter.All);
+
+        break;
+      case 'Active':
+        setFilter(Filter.Active);
+
+        break;
+      default:
+        setFilter(Filter.Completed);
+    }
+  };
+
+  const handleClearQuery = () => {
+    setQuery('');
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleSelectChange}>
+            <option value="All">All</option>
+            <option value="Active">Active</option>
+            <option value="Completed">Completed</option>
```

Comment:
Create array for these options and mapping it here, also you enum instead of words